src/DynamoCore/Library/GeometryPreloader.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using Dynamo.Interfaces;
+using DynamoUtilities;
+using ProtoScript.Runners;
+
+namespace Dynamo.Library
+{
+    /// <summary>
+    /// Geometry preloader class that helps with preloading Autodesk Shape 
+    /// Manager (ASM) binaries through geometry library (LibG). This class being
+    /// part of Dynamo core module, relies on IGeometryConfiguration supplied by
+    /// the host application to determine the installed location of ASM binaries.
+    /// </summary>
+    /// 
+    class GeometryPreloader
```

Comment:
Again, this is called `Geometry`, but is really all about ASM.  We should not hide that fact by naming things more general than they are.